cql3/type_json.cc

```diff
@@ -119,7 +119,8 @@ template <typename T> static T to_int(const rjson::value& value) {
                 "for int64 type: {} (it should not contain fractional part {})", value, fractional));
         }
 
-        if (std::numeric_limits<T>::min() > double_value || double_value > std::numeric_limits<T>::max()) {
+        if (static_cast<double>(std::numeric_limits<T>::min()) > double_value ||
```

Comment:
The warning may mean the change is wrong. What if double_value was indeed 9223372036854775808? Then the compare will fail, and we'll allow the cast, even though it doesn't fit in int64_t.

Please fix and add a unit test for int64_t::max{-1,+0,+1} and similar for the minimum value.

Reply:
s/change/original code/. The change doesn't change correctness, just silences the warning.